app/components/Reactions/ReactionPicker.tsx

```diff
@@ -0,0 +1,178 @@
+import useWindowSize from "~/hooks/useWindowSize";
+
+const EmojiPanel = React.lazy(
+  () => import("~/components/IconPicker/components/EmojiPanel")
+);
+
+type Props = {
+  /** Callback when an emoji is selected by the user. */
+  onSelect: (emoji: string) => Promise<void>;
+  /** Callback when the picker is opened. */
+  onOpen?: () => void;
+  /** Callback when the picker is closed. */
+  onClose?: () => void;
+  /** classname generated by styled-components*/
+  className?: string;
+};
+
+const ReactionPicker: React.FC<Props> = ({
+  onSelect,
+  onOpen,
+  onClose,
+  className,
+}) => {
+  const { t } = useTranslation();
+  const popover = usePopoverState({
+    modal: true,
+    unstable_offset: [0, 0],
+    placement: "bottom-end",
+  });
+
+  const { width: windowWidth } = useWindowSize();
+  const isMobile = useMobile();
+
+  const [query, setQuery] = React.useState("");
+  const contentRef = React.useRef<HTMLDivElement | null>(null);
+
+  const popoverWidth = isMobile ? windowWidth : 300;
+  // In mobile, popover is absolutely positioned to leave 8px on both sides.
+  const panelWidth = isMobile ? windowWidth - 16 : popoverWidth;
+
+  const handlePopoverButtonClick = React.useCallback(
+    (ev: React.MouseEvent) => {
+      ev.stopPropagation();
+      if (popover.visible) {
+        popover.hide();
+      } else {
+        popover.show();
+      }
+    },
+    [popover]
+  );
+
+  const handleEmojiSelect = React.useCallback(
+    (emoji: string) => {
+      popover.hide();
+      void onSelect(emoji);
+    },
+    [popover, onSelect]
+  );
+
+  // Popover open effect
+  React.useEffect(() => {
+    if (popover.visible) {
+      onOpen?.();
+    } else {
+      onClose?.();
+    }
+  }, [popover.visible, onOpen, onClose]);
+
+  // Custom click outside handling rather than using `hideOnClickOutside` from reakit so that we can
+  // prevent event bubbling.
+  useOnClickOutside(
+    contentRef,
+    (event) => {
+      if (
+        popover.visible &&
+        !popover.unstable_disclosureRef.current?.contains(event.target as Node)
+      ) {
+        event.stopPropagation();
+        event.preventDefault();
+        popover.hide();
+      }
+    },
+    { capture: true }
+  );
+
+  return (
+    <>
+      <PopoverDisclosure {...popover}>
+        {(props) => (
+          <PopoverButton
+            {...props}
+            aria-label={t("Reaction picker")}
+            className={className}
+            onClick={handlePopoverButtonClick}
+          >
+            <ReactionIcon size={20} />
+          </PopoverButton>
+        )}
+      </PopoverDisclosure>
+      {popover.visible && (
+        <Popover
+          {...popover}
+          ref={contentRef}
+          width={popoverWidth}
+          shrink
+          aria-label={t("Reaction picker")}
+          onClick={(e) => e.stopPropagation()}
+          hideOnClickOutside={false}
+        >
+          <React.Suspense fallback={<PlaceHolder />}>
```

Comment:
Honestly just `null` as placeholder might be fine, I can't manage to trigger it – even with using 3G network speeds

Reply:
I don't think we can see the placeholder in local &mdash;`EmojiPanel` component gets loaded inside the doc title's icon-picker which is lazy loaded on render. I've added the suspense (with fallback) for the concurrent access cases (for safety). A bit too much you think?

That said, here's how the placeholder looks &mdash; added an ad-hoc delay for the lazy import.

https://github.com/user-attachments/assets/7cd6075e-263b-498a-b225-2b0474deb62a